### ISSUE TITLE
The generate gzip executable fails on some gzip files while "normal" gzip happily decompresses them.

### ISSUE THREAD

**Issue by EraYaN** [ebiggers/libdeflate#31]
The used file are form the common crawl.

`https://commoncrawl.s3.amazonaws.com/crawl-data/CC-MAIN-2017-39/segments/1505818685129.23/wet/CC-MAIN-20170919112242-20170919132242-00000.warc.wet.gz` 

The files seem to have a bad size descriptor at the end, but the program (and an implementation of the library) give and error about "Not enough space in the output buffer" value 3 I think.

And the generated gzip gives `file corrupt or not in gzip format`

While OS gzip and stuff like zlib (custom implementation or gzstream) WinRar and most other seem to work just fine.

**Comment by ebiggers** [comment]
The problem is that libdeflate's gzip program doesn't support multi-member gzip files (files with multiple gzip streams concatenated to each other).  However, after looking into it, I found there is a way to support them that was not too hard.  I've pushed a fix to the branch "multi_member_gzip".  Feel free to try it, but I need to do a bit more review and testing before pushing it to the master branch.
